src/Client/LightningInternalNode.php

```diff
@@ -0,0 +1,194 @@
+    {
+        $url = $this->getApiUrl() . 'server/lightning/' .
+                    urlencode($cryptoCode) . '/info';
+
+        $headers = $this->getRequestHeaders();
+        $method = 'GET';
+
+        $response = $this->getHttpClient()->request($method, $url, $headers);
+
+        if ($response->getStatus() === 200) {
+            return new \BTCPayServer\Result\LightningNode(
+                json_decode($response->getBody(), true, 512, JSON_THROW_ON_ERROR)
+            );
+        } else {
+            throw $this->getExceptionByStatusCode($method, $url, $response);
+        }
+    }
+
+    public function connectToLightningNode(string $cryptoCode, ?string $nodeURI): bool
+    {
+        $url = $this->getApiUrl() . 'server/lightning' .
+                    urlencode($cryptoCode) . '/connect';
+
+        $headers = $this->getRequestHeaders();
+        $method = 'POST';
+
+        $body = json_encode(
+            [
+                'nodeURI' => $nodeURI,
+            ],
+            JSON_THROW_ON_ERROR
+        );
+
+        $response = $this->getHttpClient()->request($method, $url, $headers, $body);
+
+        if ($response->getStatus() === 200) {
+            return true;
+        } else {
+            throw $this->getExceptionByStatusCode($method, $url, $response);
+        }
+    }
+
+    public function getChannels(string $cryptoCode): \BTCPayServer\Result\LightningChannelList
+    {
+        $url = $this->getApiUrl() . 'server/lightning/' .
+                    urlencode($cryptoCode) . '/channels';
+
+        $headers = $this->getRequestHeaders();
+        $method = 'GET';
+
+        $response = $this->getHttpClient()->request($method, $url, $headers);
+
+        if ($response->getStatus() === 200) {
+            return new \BTCPayServer\Result\LightningChannelList(
+                json_decode($response->getBody(), true, 512, JSON_THROW_ON_ERROR)
+            );
+        } else {
+            throw $this->getExceptionByStatusCode($method, $url, $response);
+        }
+    }
+
+    public function openChannel(
+        string $cryptoCode,
+        string $nodeURI,
+        string $channelAmount,
+        int $feeRate
+    ): bool {
+        $url = $this->getApiUrl() . 'server/lightning' .
+                    urlencode($cryptoCode) . '/channels';
+
+        $headers = $this->getRequestHeaders();
+        $method = 'POST';
+
+        $body = json_encode(
+            [
+                'nodeURI' => $nodeURI,
+                'channelAmount' => $channelAmount,
+                'feeRate' => $feeRate
+            ],
+            JSON_THROW_ON_ERROR
+        );
+
+        $response = $this->getHttpClient()->request($method, $url, $headers, $body);
+
+        if ($response->getStatus() === 200) {
+            return true;
+        } else {
+            throw $this->getExceptionByStatusCode($method, $url, $response);
+        }
+    }
+
+    public function getDepositAddress(string $cryptoCode): string
```

Comment:
I tried having this return Result/Address, but threw an array expecting an array when string was returned. Is it fine just returning a string here?